### PR TITLE
fix(brewfile): add node runtime for Claude plugin hooks

### PR DESCRIPTION
**Pull request by kyriekevin**
## Summary / 概述

Phase 2 Brewfile 从未登记 `node`，导致 `brew bundle cleanup` 后 Claude Code 插件的 `.mjs` hooks（openai-codex 的 SessionStart/SessionEnd/Stop）和 `npx block-no-verify` 全部报 `/bin/sh: node: command not found`。这个 PR 把 node 显式列进 Brewfile，并同步到 README toolchain 表，让依赖可见。

Phase 2 Brewfile never declared `node`; once `brew bundle cleanup` removed it, every Claude Code plugin `.mjs` hook (openai-codex lifecycle + `npx block-no-verify`) broke with "node: command not found". This PR makes the dependency explicit in Brewfile and surfaces it in the README toolchain table.

## Change type / 变更类型

- [x] `fix` — bug fix · bug 修复

## Checklist / 检查清单

- [x] Commit messages follow Conventional Commits · Commit 信息符合 Conventional Commits
- [x] `pre-commit run --all-files` passes locally · 本地跑过 `pre-commit run --all-files` 且全绿
- [x] `chezmoi diff` reviewed; no surprising deletions or permission changes · 已 review `chezmoi diff`，没有意外的删除或权限变化
- [x] New CLI tool → added to `Brewfile` · 新 CLI 工具已加入 `Brewfile`
- [x] Smoke-tested end-to-end on at least one Mac · 在至少一台 Mac 上做过端到端冒烟

## Notes for reviewer / 给 reviewer 的说明

- **Why node and not bun** — openai-codex hooks shell out to `node` explicitly in `hooks.json`; bun has edge-case incompatibilities in ESM/stream semantics and would be a silent source of future breakage. Comment in Brewfile records the reasoning.
- **Smoke test**:
  - `node --version` → `v25.9.0`; `npx --version` → `11.12.1`
  - `node ~/.claude/plugins/cache/openai-codex/codex/1.0.3/scripts/session-lifecycle-hook.mjs SessionStart </dev/null` → exit 0 (pre-fix: `command not found`)
  - `brew bundle check --verbose` → "The Brewfile's dependencies are satisfied."
- **No test script touched** — `tests/` has no `brew.sh` (Brewfile has never had its own smoke test; `brew bundle check` in CI / apply is the de-facto check). That's also why this regression slipped through previous phases.

_Generated with Claude Code._

### CODE REVIEW

File: Brewfile [hunk 1]
```diff
@@ -44,6 +44,13 @@ brew "sevenzip"
 # ---- Python ----------------------------------------------------------------
 brew "uv"
 
+# ---- Node.js ---------------------------------------------------------------
+# Required by Claude Code plugins that ship .mjs hooks (e.g. openai-codex's
+# SessionStart/SessionEnd/Stop lifecycle hooks) and by the PreToolUse guard
+# `npx block-no-verify@1.1.2`. `bun` is not a drop-in replacement here — the
+# hooks rely on Node's native ESM/stream semantics.
+brew "node"
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply Brewfile:52]: Declining — `node@22` adds more friction than it removes for this repo's scope:

1. **Keg-only cost.** `node@22` is keg-only in Homebrew, so `brew bundle` installs it but it is **not** on PATH until you run `brew link --overwrite --force node@22`. That contradicts the `bootstrap.sh` single-shot model and would silently leave the Claude plugin hooks broken on a fresh machine — exactly the failure this PR fixes.

2. **API surface used here is stable.** The codex plugin hooks (`session-lifecycle-hook.mjs`, `stop-review-gate-hook.mjs`) use standard ESM imports + `process.argv` + `child_process.spawn` / `fs/promises` — none of that has churned across recent major node versions. `npx block-no-verify` is equally bread-and-butter. Node's semver discipline is strong; the hypothetical "major bump breaks hooks" risk is low.

3. **Global npm ecosystem assumes plain `node`.** Tools the user already uses (`npx ccusage`, `feishu-lark-mcp-server`, `bytedcli`) are installed via `npm i -g` and resolve against the default `node` symlink. Pinning to `node@22` means either manually linking or paying a PATH-precedence cost on every tool add.

4. **Reversibility is cheap.** If a future node major actually breaks a hook, pinning is a one-line Brewfile edit done in the minutes the breakage is noticed. The current cost would be permanent setup complexity for a speculative future risk — the same trade-off I declined for the cross-OS osxkeychain suggestion on PR #14.

Verification that plain `node` works today:

```
$ node --version
v25.9.0
$ node ~/.claude/plugins/cache/openai-codex/codex/1.0.3/scripts/session-lifecycle-hook.mjs SessionStart </dev/null; echo exit=$?
exit=0
$ brew bundle check --verbose --file=Brewfile
The Brewfile's dependencies are satisfied.
```

_Generated with Claude Code._

File: README.md [hunk 2]
```diff
@@ -50,6 +50,7 @@ The core tools this repo leans on.
 | [age](https://github.com/FiloSottile/age) | secret encryption | bootstrap → `brew install age` |
 | [gh](https://cli.github.com) | GitHub CLI (PR / review flow) | bootstrap → `brew install gh` |
 | [Claude Code](https://claude.com/claude-code) | AI coding CLI | bootstrap → `curl -fsSL https://claude.ai/install.sh \| bash` |
+| [Node.js](https://nodejs.org) | Runtime for Claude Code plugin hooks (`.mjs`) and `npx`-based guards | Brewfile → `brew install node` |
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply README.md:53]: Declining — the premise of this suggestion is factually wrong for this file.

Every row in the Install column of the toolchain table already uses backticks around its command:

```
L48: | [Homebrew]…  | `bootstrap.sh` (non-interactive) |
L49: | [chezmoi]…   | bootstrap → `brew install chezmoi` |
L50: | [age]…       | bootstrap → `brew install age` |
L51: | [gh]…        | bootstrap → `brew install gh` |
L52: | [Claude Code]… | bootstrap → `curl -fsSL https://claude.ai/install.sh \| bash` |
L53: | [Node.js]…   | Brewfile → `brew install node` |   ← this PR
L54: | [uv]…        | Brewfile → `brew install uv` |
L55: | [pre-commit]… | `uv tool install pre-commit` — setup in […] |
```

The Node.js row already matches the existing style. The suggested snippet (`Brewfile → brew install node@22`, no backticks) would be the **only** unbacked row in the table.

On the `node@22` half of this suggestion: see the thread on the Brewfile comment — declined there for scope / setup-friction reasons.

_Generated with Claude Code._

File: README.zh.md [hunk 3]
```diff
@@ -50,6 +50,7 @@ repo 依赖的核心工具。
 | [age](https://github.com/FiloSottile/age) | 密文加密 | bootstrap → `brew install age` |
 | [gh](https://cli.github.com) | GitHub CLI（PR / review 流程必备） | bootstrap → `brew install gh` |
 | [Claude Code](https://claude.com/claude-code) | AI coding CLI | bootstrap → `curl -fsSL https://claude.ai/install.sh \| bash` |
+| [Node.js](https://nodejs.org) | Claude Code 插件 `.mjs` hook 与 `npx` 守卫的运行时 | Brewfile → `brew install node` |
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply README.zh.md:53]: Declining — same factual issue as the English-README thread: every row in the `安装` column already uses backticks around its command (lines 48-55), so the Node.js row matches the existing convention; the suggested snippet without backticks would be the outlier.

On `node@22`: declined in the Brewfile thread — keg-only, breaks `bootstrap.sh` single-shot, conflicts with global npm tools (ccusage / feishu-lark-mcp / bytedcli).

_Generated with Claude Code._
